.ci/actor.yml

```diff
@@ -0,0 +1,112 @@
+trigger:
+  branches:
+    include:
+      - "main"
+  tags:
+    include:
+      - "kubernetes-service-applier-v*"
+  paths:
+    include:
+      - "service-applier/"
+      - ".ci/actor.yml"
+pr:
+  branches:
+    include:
+      - "main"
+  paths:
+    include:
+      - "service-applier/"
+      - ".ci/actor.yml"
+
+pool:
+  vmImage: ubuntu-latest
+
+resources:
+  repositories:
+    - repository: public-templates
+      type: github
+      endpoint: cosmonic
+      name: cosmonic/ado-common
+
+variables:
+  - group: "Common Rust Vars"
+  - group: "Cosmonic Release Keys"
+  - name: working-directory
+    value: ./service-applier
+  - name: WASH_ISSUER_KEY
+    value: $[variables.COSMONIC_ACCOUNT_OFFICIAL]
+  - name: WASH_SUBJECT_KEY
+    value: $[variables.SERVICE_APPLIER_KEY]
+
+stages:
+  - stage: build_and_check
+    jobs:
+      - job: build_and_check
+        steps:
+          - template: steps/install-wash-ubuntu.yml@public-templates
+          - template: steps/rust-setup.yml@public-templates
+            parameters:
+              components:
+                - clippy
+                - rustfmt
+              targets:
+                - wasm32-unknown-unknown
+
+          - task: Cache@2
+            inputs:
+              key: '"rust-service-applier-actor-target" | "$(Agent.OS)" | "$(working-directory)/Cargo.toml"'
+              path: $(working-directory)/target
+            displayName: Cache target folder
+
+          - task: Cache@2
+            inputs:
+              key: '"rust-service-applier-actor-cache" | "$(Agent.OS)" | "$(working-directory)/Cargo.toml"'
+              path: $(HOME)/.cargo/registry
+            displayName: Cache cargo cache folder
+
+          - template: steps/rust-test.yml@public-templates
+            parameters:
+              workingDirectory: $(working-directory)
+
+  - stage: release
+    dependsOn:
+      - build_and_check
+    # Only do this stage if the ref is a tag and the previous stage succeeded
+    condition: and(succeeded('build_and_check'), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+    jobs:
+      - job: release
+        steps:
+          - template: steps/install-wash-ubuntu.yml@public-templates
```

Comment:
This got me thinking, if the release stage only runs after the build stage, do we need to install wash/rust toolchain stuff again?

Reply:
Yeah because I believe each stage runs on a clean machine. Have not verified that though

Reply:
You're right 👍 Steps share the same filesystem/agent, but apparently jobs and stages don't share anything